UG amendments for 2101 (#176)
AY2324S1-CS2103T-T13-4/tp: docs/UserGuide.md

Step 6. Use the `java -jar FosterFamily.jar` command to run the application.<br>

>    The Foster Family GUI should appear on your screen, similar to the image below. Note how the app contains some sample data.<br>

Reviewer: "The Foster Family GUI, similar to the image below, should appear on your screen", wdyt

Author: sounds good

<br>

> To do so, enter <code>view 1</code> as shown in the image above. 

Reviewer: wait but I'm worried that if we remove the " opens the profile page of the fosterer at index 1, Alex Yeoh" it might be unclear to her what does 1 mean

Author: true :O ill add it back!

Author: wait! actually at the top, just before the image, it was already mentioned. thats why i deleted it!
<img width="447" alt="Screenshot 2023-11-15 at 3 26 01 PM" src="https://github.com/AY2324S1-CS2103T-T13-4/tp/assets/111829949/7ba64269-0158-43e1-b6c4-a64485f7512f">

wdyt!

Reviewer: lgtm, sorry I didnt see that haha

<div style="page-break-after: always;"></div>

#### Navigating through fields

> Entering the full name or substring of the field you want to edit brings your cursor to the corresponding textbox.

Reviewer 2: ```suggestion
Entering the name, fully or partially, of the field you want to edit brings your cursor to the corresponding textbox.
```

<div style="page-break-after: always;"></div>

> After you are **done editing**, _press the Enter key_ on your keyboard to bring your cursor back to the command box. 

Reviewer 2: ```suggestion
After you are done editing, _press the Enter key_ on your keyboard to **confirm your changes**, and bring your cursor back to the command box. 
```

<div></div>

> If you **wish to discard changes** while editing in the textbox, _press the Esc key_ to undo the changes and direct your cursor back to the command box.

Reviewer 2: ```suggestion
If you wish to **discard changes** instead, _press the Esc key_. This reverts your most recent changes, and directs your cursor back to the command box.
```

Author: i feel like this might be misinterpreted as like all recent changes, rather than the revert being specific to that one field though. Ill clarify the phrasing

Reviewer 2: That's true. Maybe we can mention something about unconfirmed changes? Since other fields' changes would be confirmed.

Author: that may bring about the need to define what is confirmed and what is unconfirmed though! u see if u wanna add it in ur PR, ill leave it as is first

<br/>

If this was an `edit`, your changes to the fosterer will be saved.<br>

> If this was an `add`,  the new fosterer with the inputted details will be added.

Reviewer: for add I think can add that "you will exit the profile page and the new fosterer.....will be added"

> If this was an `add`,  the new fosterer with the inputted details will be added.

Reviewer 2: ```suggestion
The effect of `save` depends on the command you used to enter this view.
If this was an `edit`, your changes to the fosterer will be saved.<br>
If this was an `add`,  the new fosterer with the inputted details will be added.
```

Author: can ill add this in, sorry this was outdated

<br>

#### Exiting the profile page

Entering <code>exit</code> closes the profile page and directs you back to the main window. <br>

> Foster Family will warn you if you attempt to exit without first saving your changes.

Reviewer: Don't need "first" here, sounds off

Author: all right~

<div></div>

* `list "John" / zam & doe`
    * lists fosterers which match "John Doe" and "Doe Shazam", but not "John Grahm".

<br> 

<div style="page-break-after: always;"></div>

### Viewing a fosterer's detail: `view`

> Redirects you to the profile page of the fosterer at the specified index in your address book.

Reviewer 2: ```suggestion
Redirects you to the profile page of the fosterer at the specified index of the currently displayed list.
```
Not too sure about my own suggestion, but I think the idea that it's affected by `list` should be relevant here.

Author: can, will add it back for the other ones

**:exclamation: Important:**<br>

Only the <code>save</code> and <code>exit</code> commands are available to you in the profile page.

> </div> 

Reviewer 2: Maybe the "learn more" part below should be included here?

**:exclamation: Important:**<br>

* If the parameters are not provided, <b><code>edit INDEX</code> operates the same way as <code>view INDEX</code></b>, leading you to the profile page of the person at index <code>INDEX</code> in the address book. 
* If you run the **same edit command multiple times consecutively** (resulting in no visible change after the first run), the undo command **will not** be able to revert the data back to the original state. <br>

>   This is because undo can only undo the last command, **even if the command made no visible changes**.

Reviewer: In the sentence "this is because...", can we make the "undo" appear in code format, like the code `undo` thing HAHAHA, same for a similar sentence in sort i think

Author: can!

>   This is because undo can only undo the last command, **even if the command made no visible changes**.

Reviewer 2: Thus far, I think bold is for user intentions, and italics is for general purpose emphasis? So perhaps these should be italics?

### Editing a fosterer's details through the profile page: `edit`

> Edits the details of the fosterer at the specified index in your address book, done through the profile page.

Reviewer 2: ```suggestion
Edits the details of the fosterer at the specified index of the currently displayed list, by redirecting to the profile page.
```

**:warning: Caution:**<br>

Edit may cause **information loss**. 

> Before you type the save command, please check that the details of the fosterer are correct.

Reviewer 2: Might be a bit awkward, since edits edit.
Not an issue, I just think it might be a bit extra?

Author: can! removed

</div>

<br/>

If you have at least one fosterer in your address book, here is an example of a profile page you will see after entering `view 1` or `edit 1`:

<div style="text-align: center">
<img src="images/screenshots/View1.png" height="480" class="center"/>
</div>

To learn more about the profile page, please refer to the section [User Interface: The Profile Page](#the-profile-page). 

<br> 

<div style="page-break-after: always;"></div>

### Saving changes in a fosterer's details: `save`

Saves changes to details of the fosterer which you have made in the profile page.

<div markdown="block" class="alert alert-danger">

**:exclamation: Important:**<br>

This command only works in the profile page, which you can navigate to by executing the `view` command.

</div>

Format: `save`

<div markdown="block" class="alert alert-info">

**:information_source: Notes about the command:**<br>

*  Entering <code>save</code> in [the profile of a new fosterer](#adding-a-fosterer-through-the-profile-page-add) saves the new fosterer and exits the profile page.

> *  Entering <code>save</code> in [the profile of an already existing fosterer](#editing-a-fosterers-details-through-the-profile-page-edit) saves the changes but does not exit the profile page, in case you want to edit more details.

Reviewer: instead of "in case you want to edit more details" I think smth like "so that you can continue editing more details" sounds more formal, wdyt

</div>

<br> 

<div style="page-break-after: always;"></div>

### Deleting a fosterer : `delete`

> Deletes the fosterer at the specified index in your address book.

Reviewer 2: ```suggestion
Deletes the fosterer at the specified index of the currently displayed list.
```

On the profile page,<br>

* Entering <code>exit</code> leads you back to the main window. 

> * Attempting to <code>exit</code> **without first saving changes** with <code>save</code> prompts a warning message.

Reviewer: same for this, "without saving changes", I think it sounds more natural without a "first" there

> * Please refer to [User Interface: The Profile Page: Exiting the profile page](#exiting-the-profile-page) for more information. 

Reviewer 2: ```suggestion
On the profile page, entering <code>exit</code> leads you back to the main window, instead of exiting the application. 
Please refer to [User Interface: The Profile Page: Exiting the profile page](#exiting-the-profile-page) for more information. 
```

</div>
 

### Saving data

In the **main window**, your Foster Family data is **saved in the hard disk automatically** after any command that changes the data, so no manual saving is needed. However, **edits made in the profile page have to be saved** via the `save` command. Else, changes will be discarded once you exit out of that fosterer's profile page.

<div style="page-break-after: always;"></div>

### Editing data file

* Your Foster Family data is saved automatically as a JSON file `[JAR file location]/data/addressbook.json`.<br>

* A JSON file stores fosterer details as key value pairs, making it more readable than a regular text file.<br>

  ![Json](images/screenshots/JsonExample.png)

>   For example, the "name" key is paired with a value "Alex Yeoh" for the first fosterer in your address book.<br>

Reviewer 2: If we are not recommending users to edit it, perhaps we don't need to teach them about JSON? maybe [an external link](https://www.w3schools.com/whatis/whatis_json.asp) would make sense.

Author: I still feel that it is better to clarify what a json is though. Not encouraging them to edit!! viewing the json may not mean that they are gonna edit, they may need it to copy over data to another computer etc, so at least they should know what its about (at least thats what i think! wdyt of that)

Author: if u wanna, you can make the do not edit message  clearer here

Reviewer 2: That's fair. I think it's clear enough that editing is discouraged.

## **Known issues**

> - **When using multiple screens**, if you move the application to a secondary screen, and later switch to using only the primary screen, the GUI will open off-screen. The solution is to delete the `preferences.json` file created by the application before running the application again.

Reviewer: actually i nvr understood what does off-screen mean HAHAH but i think this is from the original ab3?

Author: yeaa it was from the original one. i think it may mean like a part of the window may be cut off once u switch back to using ur main screen only. ill make that part clearer

Reviewer: thanks!
